test/unit/AccountingModule.t.sol

```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.28;
+
+import { Test, console } from "forge-std/Test.sol";
+import { MockERC20 } from "../mocks/MockERC20.sol";
+import { MockStrategy } from "../mocks/MockStrategy.sol";
+import { AccountingModule, IAccountingModule } from "../../src/AccountingModule.sol";
+import { AccountingToken } from "../../src/AccountingToken.sol";
+
+contract AccountingModuleTest is Test {
```

Comment:
can we enhance the suite with fuzzing as well?

Can be additional tests that do the fuzzing on deposited amounts, rewards, withdrawn amounts